docs/guide/exe_verilator.rst

```diff
@@ -733,6 +733,13 @@ Summary:
    :option:`/*verilator&32;hier_block*/` metacomment is ignored.  See
    :ref:`Hierarchical Verilation`.
 
+.. option:: --hierarchical-dpi-cost <cost>,<name>
+
+   Internal flag inserted used during :vlopt:`--hierarchical`; specifies
+   execution cost of a hierarchical DPI wrappers for modules with
+   :option:`/*verilator&32;hier_block*/` metacomment. See
+   :ref:`Hierarchical Verilation`.
+
```

Comment:
I'd suggest we instead use the profile_data in the .vlt file format to communicate this, you can see an example writer in verilated_profiler.h

These profile_data vlt comments can be embedded directly into the generated .v file e.g..

     ... verilog ...
```suggestion
    --hierarchical-dpi-cost <cost>,<name> Internal option that specifies DPI function cost
     `verilator_config
     profile_data -model ... -mtask ... -cost 64'd234
     `verilog
```

We might need some new option rather than -mtask.

I prefer this as we can then in the future enhance the profiling code to collect the real delays of the functions, which will give you even better performance.  Also, otherwise, there are command line length limits which we could hit on large designs.